File: Bai_12/main.cpp
#include <stdio.h>

void findMinMax(int *arr, int size, int *min, int *max) {
    *min = *max = arr[0];
    for (int i = 1; i < size; ++i) {
        if (arr[i] < *min) {
            *min = arr[i];
        }
        if (arr[i] > *max) {
            *max = arr[i];
        }
    }
}

int main() {
    int size;
    printf("Nhập kích thước của mảng: ");
    scanf("%d", &size);

    int arr[size];
    printf("Nhập các phần tử của mảng:\n");
    for (int i = 0; i < size; ++i) {
        scanf("%d", &arr[i]);
    }

    int min, max;
    findMinMax(arr, size, &min, &max);

    printf("Giá trị nhỏ nhất của mảng: %d\n", min);
    printf("Giá trị lớn nhất của mảng: %d\n", max);

    return 0;
}

File: Bai_12/main.py
def find_min_max(arr):
    min_val = max_val = arr[0]
    for num in arr[1:]:
        if num < min_val:
            min_val = num
        if num > max_val:
            max_val = num
    return min_val, max_val

if __name__ == "__main__":
    size = int(input("Nhập kích thước của mảng: "))
    arr = []
    print("Nhập các phần tử của mảng:")
    for _ in range(size):
        arr.append(int(input()))

    min_val, max_val = find_min_max(arr)

    print("Giá trị nhỏ nhất của mảng:", min_val)
    print("Giá trị lớn nhất của mảng:", max_val)
